repository-ydb-v2/src/main/java/tech/ydb/yoj/repository/ydb/table/YdbTable.java

```diff
@@ -54,18 +57,36 @@
 import static tech.ydb.yoj.repository.db.EntityExpressions.defaultOrder;
 
 public class YdbTable<T extends Entity<T>> implements Table<T> {
+    private static final AtomicBoolean useOldStatementFactory = new AtomicBoolean(Boolean.getBoolean("yoj.use.type.statement.factory"));
+
+    @Deprecated(forRemoval = true)
+    public static void setUseOldStatementFactory(boolean value) {
+        DeprecationWarnings.warnOnce("YdbTable.setUseOldStatementFactory(boolean)",
+                "You are using YdbTable.setUseOldStatementFactory(boolean) which is deprecated for removal in YOJ 3.0.0. Please stop calling this method");
+        useOldStatementFactory.set(value);
+    }
+
     private final QueryExecutor executor;
     @Getter
-    private final Class<T> type;
+    private final EntityDescriptor<T> descriptor;
```

Comment:
Nobody seems to use `getDescriptor()` which is generated by the `@Getter` annotation on the `descriptor` field. Let's remove this annotation for now (and the `getType()` method will be OK, it's no longer generated by Lombok)

Reply:
Removed getter annotation